README.md

```diff
@@ -115,6 +117,44 @@ The promise **must** return a response object anything else will lead to a 'HTTP
 
 Other types aren't allowed and will lead to a 'HTTP 500 Internal Server Error' response for the client.
 
+### Middleware
+
+#### Creating your own middleware
+
+You can create your own middleware, which lies between the server and the callback function. Use this middleware to manipulate the requests or
+responses of the server. You can add as many middlewar as you want you just need to follow the following design
+
+```php
+$callback = function (RequestInterface $request) {
+    return new Response();
+}
+
+$middleware = function (RequestInterface $request, callable callables) {
```

Comment:
typo here.

Also, what is being passed here and why? Looks like an array of callables, in which case this can not be type casted? Is *every* middleware supposed to pop from this array? Why is this not a single callable?